metrics/core/lib/src/util/validation/printer/validation_result_printer.dart

```diff
@@ -0,0 +1,135 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:metrics_core/src/util/validation/target_validation_result.dart';
+import 'package:metrics_core/src/util/validation/validation_conclusion.dart';
+import 'package:metrics_core/src/util/validation/validation_result.dart';
+import 'package:metrics_core/src/util/validation/validation_target.dart';
+
+/// A class that is responsible for printing the [ValidationResult].
+class ValidationResultPrinter {
+  /// An instance of the [StringSink] this printer uses to print the result.
+  final StringSink sink;
+
+  /// Create a new instance of the [ValidationResultPrinter] with the given
+  /// [StringSink].
+  ///
+  /// If the [StringSink] is `null`, an instance of the standard output stream
+  /// is used.
+  ValidationResultPrinter({
+    StringSink sink,
+  }) : sink = sink ?? stdout;
+
+  /// Prints the given [result] to the [sink].
+  void print(ValidationResult result) {
+    final validationResults = result.results;
+
+    final validationResultEntries = validationResults.entries;
+
+    for (final entry in validationResultEntries) {
+      final target = entry.key;
+      final result = entry.value;
+
+      final message = _buildTargetValidationOutput(target, result);
+
+      sink.writeln(message);
+    }
+  }
+
+  /// Builds the output that provides the information about
+  /// the given [target] and [validationResult].
+  String _buildTargetValidationOutput(
+    ValidationTarget target,
+    TargetValidationResult validationResult,
+  ) {
+    final conclusion = _getConclusionIndicator(validationResult);
+    final targetName = target.name;
+    final targetDescription = _getDescription(target);
+    final validationDescription = _getValidationDescription(validationResult);
+    final details = _getValidationDetails(validationResult);
+    final context = _getValidationContext(validationResult);
+
+    return '''
+$conclusion $targetName$targetDescription$validationDescription$details
+\t$context
+    ''';
```

Comment:
:todo:
It seems like we can refactor this a little to simplify the resulting string and the structure at all. Please DM me for more details.